sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -206,7 +206,25 @@ protected override async Task<bool> AcceptCommand (MessageId id, string method,
 							break;
 							}
 						case "array": {
-							await GetDetails (id, MonoCommands.GetArrayValues (int.Parse (parts [2])), token);
+							switch (parts.Length) {
+							case 3: {
+								await GetDetails (id, MonoCommands.GetArrayValues (int.Parse (parts [2])), token);
+								break;
+							}
+							case 4: {
+								// This form of the id is being used only for valuetypes right now
+								if (!IsValueTypeCached (id, objId)) {
+									var arrayObjectId = int.Parse (parts [2]);
+									var idx = int.Parse (parts [3]);
+									await GetDetails (id, MonoCommands.GetArrayValueExpanded (arrayObjectId, idx), token, send_response: false);
+								}
+								GetDetailsForValueType (id, objId, token);
+								break;
+							}
+							default:
+								throw new ArgumentException ($"Unknown objectId format for array: {objId}");
```

Comment:
We don't really want the proxy throwing uncaught exceptions.  `Runtime.getProperties` has exceptionDetails in the result so let's fill that in instead.   Also please move this out of the switch and into a function.